tests/test_kb_dataset.py

```diff
@@ -0,0 +1,127 @@
+import argparse
+import sys
+import unittest
+from collections import defaultdict
+from pathlib import Path
+
+import datasets
+
+
+
+class TestKBDataset(unittest.TestCase):
+    def setUp(self) -> None:
+        self.dataset_source = datasets.load_dataset(sys.argv[1], name="source")
+        self.dataset_bigbio = datasets.load_dataset(sys.argv[1], name="bigbio")
+
+    def print_statistics(self):
+        for split_name, split in self.dataset_bigbio.items():
+            print(split_name)
+            print("=" * 10)
+
+            counter = defaultdict(int)
+            for example in split:
+                for feature_name, feature in features.items():
+                    if isinstance(feature, datasets.Value) or isinstance(
+                        feature_name, dict
+                    ):
+                        if example[feature_name]:
+                            counter[feature_name] += 1
```

Comment:
If I'm understanding this conditional correctly, I'd remove it. I think we can allow for some top level keys being missing if the data doesn't support it (i.e. no coreferences if they are not in the data). but i think we should enforce that every example has a given key if that key is included in the dataset. (i might be mis-reading these `for` loops though)

Reply:
ok got it, yea, that makes sense. fine with leaving for now.